pynucastro/networks/rate_collection.py

```diff
@@ -693,6 +693,19 @@ def evaluate_rates(self, rho, T, composition, screen_func=None):
 
         return rvals
 
+    def find_unimportant_rates(self, rhos, Ts, compositions, cutoff_ratio,
+                               screen_func=None):
```

Comment:
if I understand this correctly, it is doing the filtering in each thermodynamic state separately.
But for a full network run, we want to keep a rate if it is every important, which means that we want to compare the fastest a particular rate is over all thermodynamic conditions to the cuttoff threshold.

Reply:
It only does the filtering at the end, after finding the maximum ratio for each rate over all the thermodynamic states. In the tests, rate 3 is unimportant at T=1e8, while it is important at T=1e10; vice versa for rate 0. In the third test that includes both T=1e8 and 1e10, both 0 and 3 are excluded from the unimportant rates. I'll update the tests to be more explicit about this

Reply:
ah okay.  I need to read it more carefully then

Reply:
oh, I know why I was confused.  You are still computing the ratio based on the fastest at that thermodynamic state.  I was thinking of there being a global fastest across all states that we compare to.  But I think that your method is a little more conservative, and probably better.